zendriver/core/browser.py

```diff
@@ -595,7 +606,7 @@ async def _cleanup_temporary_profile(self) -> None:
 
         for attempt in range(5):
             try:
-                shutil.rmtree(self.config.user_data_dir, ignore_errors=False)
+                shutil.rmtree(str(self.config.user_data_dir), ignore_errors=False)
```

Comment:
I don't think it's necessary since we already do this in the Config setter

```suggestion
                shutil.rmtree(self.config.user_data_dir, ignore_errors=False)
```

Reply:
It is needed because we'll get a `mypy` error otherwise. This just makes using it as a `str` explicit since the `user_data_dir` is stored as a `str` internally but the `rmtree` method allows more than that.
```
zendriver/core/browser.py:610: error: Argument 1 to "__call__" of "_RmtreeType" has incompatible type "str | None"; expected "str | bytes | PathLike[str] | PathLike[bytes]"  [arg-type]
```

Reply:
Let me take a look. The problem here is that the internal variable is Optional but the public property will not be after the changes I've made. Let me take another look at this though with this new perspective.

Reply:
Okay, I settled on an approach that tricks `mypy` enough to pass review. 🤣 This is admittedly a bit of a strange case, but I was originally wrestling with `mypy` not having the forward declaration of the variable in the init, etc. I'll put up another commit and can squash everything back down when all reviewed.